plugin/src/main/java/com/google/errorprone/xplat/checker/JodaTimeClassBan.java

```diff
@@ -0,0 +1,269 @@
+ * org.joda.time.Months, org.joda.time.MutableDateTime, org.joda.time.MutableInterval,
+ * org.joda.time.MutablePeriod, org.joda.time.Partial, org.joda.time.PeriodType,
+ * org.joda.time.ReadWritableDateTime, org.joda.time.ReadWritableInstant,
+ * org.joda.time.ReadWritableInterval, org.joda.time.ReadWritablePeriod org.joda.time.Seconds,
+ * org.joda.time.TimeOfDay org.joda.time.Weeks org.joda.time.YearMonthDay org.joda.time.Years}.
+ * These calls are banned due to their incompatibility with cross platform development.
+ */
+@BugPattern(
+    name = "JodaTimeClassBan",
+    summary = "Joda Time cross platform class ban",
+    explanation =
+        "The usage of several Joda Time classes are banned from cross platform development due"
+            + " to incompatibilities. They are unsupported on the web and should also not be"
+            + " used on supported platforms.",
+    severity = ERROR)
+public class JodaTimeClassBan extends BugChecker
+    implements MethodInvocationTreeMatcher, NewClassTreeMatcher, ImportTreeMatcher,
+    VariableTreeMatcher, MethodTreeMatcher {
+
+  private static final ImmutableSet<String> PACKAGE_NAMES =
+      ImmutableSet.of(
+          "org.joda.time.base.", "org.joda.time.chrono", "org.joda.time.convert",
+          "org.joda.time.field", "org.joda.time.format", "org.joda.time.tz"
+      );
+
+  private static final ImmutableSet<String> CLASS_NAMES =
+      ImmutableSet.of(
+          "org.joda.time.Chronology", "org.joda.time.DateMidnight",
+          "org.joda.time.DateTimeComparator", "org.joda.time.DateTimeField",
+          "org.joda.time.DateTimeFieldType", "org.joda.time.DateTimeUtils",
+          "org.joda.time.Days", "org.joda.time.DurationField", "org.joda.time.DurationFieldType",
+          "org.joda.time.Hours", "org.joda.time.IllegalFieldValueException",
+          "org.joda.time.IllegalInstantException", "org.joda.time.JodaTimePermission",
+          "org.joda.time.Minutes", "org.joda.time.MonthDay", "org.joda.time.Months",
+          "org.joda.time.MutableDateTime", "org.joda.time.MutableInterval",
+          "org.joda.time.MutablePeriod", "org.joda.time.Partial", "org.joda.time.Years",
+          "org.joda.time.PeriodType", "org.joda.time.ReadWritableDateTime",
+          "org.joda.time.ReadWritableInstant", "org.joda.time.ReadWritableInterval",
+          "org.joda.time.ReadWritablePeriod", "org.joda.time.Seconds",
+          "org.joda.time.TimeOfDay", "org.joda.time.Weeks", "org.joda.time.YearMonthDay"
+      );
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+
+    Symbol methodSymbol = ASTHelpers.getSymbol(tree);
+    Type methodRecvType = ASTHelpers.getReceiverType(tree);
+    Type methodType = ASTHelpers.getType(tree);
+
+    if (methodType != null && methodSymbol != null) {
+
+      //checks receiver
+      if (CLASS_NAMES.contains(methodRecvType.toString())) {
+        return buildDescription(tree)
```

Comment:
Seems there are only two types of messages for the `Description`s we return:

`Use of %s has been banned...`

asd

`Use of %s is not allowed, as %s has been banned...`

What do you think about extracting them to two helper private methods, so that you don't have to repeat them below?

Reply:
Sounds good.